heartbeat/componentHandler/heartbeatHandler.go

```diff
@@ -0,0 +1,274 @@
+		HardforkTrigger:  arg.HardforkTrigger,
+	}
+
+	hbh.sender, err = process.NewSender(argSender)
+	if err != nil {
+		return err
+	}
+
+	log.Debug("heartbeat's sender component has been instantiated")
+
+	heartBeatMsgProcessor, err := process.NewMessageProcessor(
+		arg.SingleSigner,
+		arg.KeyGenerator,
+		arg.Marshalizer,
+		arg.PeerShardMapper,
+	)
+	if err != nil {
+		return err
+	}
+
+	heartbeatStorer, err := heartbeatStorage.NewHeartbeatDbStorer(arg.Storer, arg.Marshalizer)
+	if err != nil {
+		return err
+	}
+
+	timer := &process.RealTimer{}
+	netInputMarshalizer := arg.Marshalizer
+	if arg.SizeCheckDelta > 0 {
+		netInputMarshalizer = marshal.NewSizeCheckUnmarshalizer(arg.Marshalizer, arg.SizeCheckDelta)
+	}
+
+	allValidators, _, _ := hbh.getLatestValidators()
+	pubKeysMap := make(map[uint32][]string)
+	for shardID, valsInShard := range allValidators {
+		for _, val := range valsInShard {
+			pubKeysMap[shardID] = append(pubKeysMap[shardID], string(val.PublicKey))
+		}
+	}
+
+	argMonitor := process.ArgHeartbeatMonitor{
+		Marshalizer:                        netInputMarshalizer,
+		MaxDurationPeerUnresponsive:        time.Second * time.Duration(arg.HeartbeatConfig.DurationInSecToConsiderUnresponsive),
+		PubKeysMap:                         pubKeysMap,
+		GenesisTime:                        arg.GenesisTime,
+		MessageHandler:                     heartBeatMsgProcessor,
+		Storer:                             heartbeatStorer,
+		PeerTypeProvider:                   peerTypeProvider,
+		Timer:                              timer,
+		AntifloodHandler:                   arg.AntifloodHandler,
+		HardforkTrigger:                    arg.HardforkTrigger,
+		PeerBlackListHandler:               arg.PeerBlackListHandler,
+		ValidatorPubkeyConverter:           arg.ValidatorPubkeyConverter,
+		HbmiRefreshIntervalInSec:           arg.HeartbeatConfig.HbmiRefreshIntervalInSec,
+		HideInactiveValidatorIntervalInSec: arg.HeartbeatConfig.HideInactiveValidatorIntervalInSec,
+	}
+	hbh.monitor, err = process.NewMonitor(argMonitor)
+	if err != nil {
+		return err
+	}
+
+	log.Debug("heartbeat's monitor component has been instantiated")
+
+	err = hbh.monitor.SetAppStatusHandler(arg.AppStatusHandler)
+	if err != nil {
+		return err
+	}
+
+	err = arg.Messenger.RegisterMessageProcessor(core.HeartbeatTopic, hbh.monitor)
+	if err != nil {
+		return err
+	}
+
+	hbh.ctx, hbh.cancelFunc = context.WithCancel(context.Background())
+	go hbh.startSendingHeartbeats()
+
+	return nil
+}
+
+func (hbh *HeartbeatHandler) getLatestValidators() (map[uint32][]*state.ValidatorInfo, map[string]*state.ValidatorApiResponse, error) {
+	latestHash, err := hbh.arg.ValidatorStatistics.RootHash()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	validators, err := hbh.arg.ValidatorStatistics.GetValidatorInfoForRootHash(latestHash)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return validators, nil, nil
+}
+
+func (hbh *HeartbeatHandler) startSendingHeartbeats() {
```

Comment:
rename to sendingHeartBeats ?

Reply:
it was easier to implement this as the difference between 2 timestamps is expressed in nanoseconds.